src/Microsoft.ML.ImageAnalytics/ImageLoader.cs

```diff
@@ -85,6 +91,21 @@ internal ImageLoadingTransformer(IHostEnvironment env, string imageFolder = null
             : base(Contracts.CheckRef(env, nameof(env)).Register(nameof(ImageLoadingTransformer)), columns)
         {
             ImageFolder = imageFolder;
+            UseImageType = true;
+        }
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="ImageLoadingTransformer"/>.
+        /// </summary>
+        /// <param name="env">The host environment.</param>
+        /// <param name="imageFolder">Folder where to look for images.</param>
+        /// <param name="type">Image type flag - true for ImageDataViewType or false for VectorDataViewType. Defaults to true i.e. ImageDataViewType if not specified.</param>
+        /// <param name="columns">Names of input and output columns.</param>
+        internal ImageLoadingTransformer(IHostEnvironment env, string imageFolder = null, bool type = true, params (string outputColumnName, string inputColumnName)[] columns)
```

Comment:
>true [](start = 102, length = 4)

I don't see the advantage of having a default of `true` here, since there is already a constructor where this parameter is not specified and it defaults to true.

Reply:
Actually, this doesn't need to have a default at all, since whenever the `type` parameter is not specified, it will call the other ctor.

---
In reply to: [330462595](https://github.com/dotnet/machinelearning/pull/4242#discussion_r330462595) [](ancestors = 330462595)